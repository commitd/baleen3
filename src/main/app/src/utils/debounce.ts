/*-
 * #%L
 * Baleen 3
 * %%
 * Copyright (C) 2020 Dstl
 * %%
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * #L%
 */
/* eslint-disable @typescript-eslint/no-explicit-any */
export type Method = (...args: any[]) => void

export function debounce<F extends Method>(
  func: F,
  wait = 200
): (this: ThisParameterType<F>, ...args: Parameters<F>) => void {
  let timeoutId: ReturnType<typeof setTimeout> | undefined

  return function (this: ThisParameterType<F>, ...args: Parameters<F>): void {
    // eslint-disable-next-line @typescript-eslint/no-this-alias
    const context = this

    if (timeoutId !== undefined) {
      clearTimeout(timeoutId)
    }
    timeoutId = setTimeout(() => {
      timeoutId = undefined
      func.apply(context, args)
    }, wait)
  }
}
