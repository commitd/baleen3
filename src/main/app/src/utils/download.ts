/*-
 * #%L
 * Baleen 3
 * %%
 * Copyright (C) 2020 Dstl
 * %%
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * #L%
 */
// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const downloadJson = (name: string, data: any): void => {
  download(
    name + '.json',
    'data:text/json;charset=utf-8,',
    JSON.stringify(data, null, 2)
  )
}

export const download = (
  name: string,
  mediaType: string,
  data: string
): void => {
  const dataStr = mediaType + encodeURIComponent(data)
  const downloadAnchorNode = document.createElement('a')
  downloadAnchorNode.setAttribute('href', dataStr)
  downloadAnchorNode.setAttribute('download', name)
  document.body.appendChild(downloadAnchorNode) // required for firefox
  downloadAnchorNode.click()
  downloadAnchorNode.remove()
}
