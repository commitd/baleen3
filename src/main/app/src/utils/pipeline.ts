/*-
 * #%L
 * Baleen 3
 * %%
 * Copyright (C) 2020 Dstl
 * %%
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * #L%
 */
import Ajv from 'ajv'
import { v4 } from 'uuid'
import {
  Annot8ComponentDescriptor,
  ComponentInfo,
  ComponentMap,
  ComponentMetadata,
  PipelineDescriptor,
  PipelineEditDescriptor,
  PipelineViewDescriptor,
  ServerDetails,
  SettingsMap,
  SettingsSchema,
} from '../types'

const ajv = new Ajv({ useDefaults: true })
// eslint-disable-next-line @typescript-eslint/no-unsafe-call, @typescript-eslint/no-unsafe-return
const uuid = (): string => v4()

export const isSettingsValid = (
  schema?: SettingsSchema,
  settings?: object
): boolean => {
  if (schema === undefined) {
    // No settings schema, assume valid
    return true
  }

  try {
    const jsonSchema = JSON.parse(schema.jsonSchema) as object
    const validator = ajv.compile(jsonSchema)
    const valid = validator(settings) as boolean
    if (!valid) {
      let message = 'JSON not valid:'
      if (!(validator.errors == null)) {
        validator.errors.forEach((error: Ajv.ErrorObject): void => {
          message += error.message
        })
      }
      console.debug(message)
    }
    return valid
  } catch (e) {
    console.debug(e)
    return false
  }
}

export const defaultSettings = (
  schema?: SettingsSchema
): object | undefined => {
  if (schema === undefined) {
    // No settings schema, assume no settings
    return undefined
  }

  try {
    const jsonSchema = JSON.parse(schema.jsonSchema) as object
    const validator = ajv.compile(jsonSchema)
    const defaultData = {}
    const valid = validator({}) as boolean
    if (valid) {
      return defaultData
    }
  } catch (e) {
    console.debug(e)
  }
  return undefined
}

export const isAllComponentsValid = (
  components: ComponentMetadata[]
): boolean => {
  return components.find((c) => !c.valid) === undefined
}

export const isPipelineValid = (
  nameValid: boolean,
  pipeline: Readonly<PipelineViewDescriptor>
): string[] => {
  const errors = []
  if (!nameValid) {
    errors.push('The pipeline name must be unique')
  }
  if (pipeline.sourceOrder.length === 0) {
    errors.push('A pipeline must have at least one source')
  }
  if (pipeline.processorOrder.length === 0) {
    errors.push('A pipeline must have at least one processor')
  }
  if (!isAllComponentsValid(Object.values(pipeline.components))) {
    errors.push('All components must be valid')
  }

  return errors
}

export const createComponent =
  (settings: SettingsMap, types: ComponentMap, configured = true) =>
  (descriptor: Annot8ComponentDescriptor): ComponentMetadata => {
    const id = uuid()
    const type = Object.keys(descriptor)[0]
    const info: ComponentInfo | undefined = types[`${type}`]
    const schema: SettingsSchema | undefined =
      info !== undefined && info.settingsClass !== undefined
        ? settings[info.settingsClass]
        : undefined

    if (schema !== undefined && descriptor[`${type}`].settings === undefined) {
      descriptor[`${type}`].settings = defaultSettings(schema)
    }

    return {
      id,
      descriptor,
      info,
      schema,
      valid:
        info !== undefined &&
        isSettingsValid(schema, descriptor[`${type}`].settings),
      configured,
    }
  }

export const createPipelineViewDescriptor = (
  pipeline: PipelineDescriptor,
  serverDetails: ServerDetails
): PipelineViewDescriptor => {
  const components: { [key: string]: ComponentMetadata } = {}
  const sourceOrder: string[] = []
  const processorOrder: string[] = []

  pipeline.sources
    .map(createComponent(serverDetails.settings, serverDetails.sources))
    .forEach((component) => {
      components[component.id] = component
      sourceOrder.push(component.id)
    })

  pipeline.processors
    .map(createComponent(serverDetails.settings, serverDetails.processors))
    .forEach((component) => {
      components[component.id] = component
      processorOrder.push(component.id)
    })

  return {
    name: pipeline.name,
    description: pipeline.description,
    sourceOrder,
    processorOrder,
    components,
    errorConfiguration: pipeline.errorConfiguration,
  }
}

export const createPipelineEditDescriptor = (
  pipeline: PipelineDescriptor,
  orderer: string,
  usedNames: string[],
  serverDetails: ServerDetails
): PipelineEditDescriptor => {
  const pipelineView = createPipelineViewDescriptor(pipeline, serverDetails)

  const nameValid = !usedNames.includes(pipeline.name)

  return {
    ...pipelineView,
    orderer,
    nameValid,
    errors: isPipelineValid(nameValid, pipelineView),
  }
}
