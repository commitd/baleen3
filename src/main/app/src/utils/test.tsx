/*-
 * #%L
 * Baleen 3
 * %%
 * Copyright (C) 2020 Dstl
 * %%
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * #L%
 */
import { StylesProvider } from '@material-ui/styles'
import {
  createHistory,
  createMemorySource,
  History,
  LocationProvider,
} from '@reach/router'
import { render, RenderOptions, RenderResult } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { GenerateId } from 'jss'
import React from 'react'
import { ThemeProvider } from '../components/ThemeProvider/ThemeProvider'

const generateClassName: GenerateId = (rule, styleSheet) => {
  const prefix =
    styleSheet === undefined || styleSheet.options.classNamePrefix === undefined
      ? ''
      : styleSheet.options.classNamePrefix
  return `${prefix}-${rule.key}`
}

const LightTheme: React.FC<{ readonly children?: React.ReactNode }> = ({
  children,
}) => (
  <StylesProvider generateClassName={generateClassName}>
    <ThemeProvider choice="light">{children}</ThemeProvider>
  </StylesProvider>
)

const DarkTheme: React.FC<{ readonly children?: React.ReactNode }> = ({
  children,
}) => (
  <StylesProvider generateClassName={generateClassName}>
    <ThemeProvider choice="dark">{children}</ThemeProvider>
  </StylesProvider>
)

export const renderPlain = render

export const renderLight = (
  ui: Readonly<React.ReactElement>,
  options?: Omit<RenderOptions, 'queries'>
): RenderResult => render(ui, { wrapper: LightTheme, ...options })

export const renderDark = (
  ui: Readonly<React.ReactElement>,
  options?: Omit<RenderOptions, 'queries'>
): RenderResult => render(ui, { wrapper: DarkTheme, ...options })

export function renderWithRouter(
  ui: Readonly<React.ReactElement>,
  route = '/',
  options?: Omit<RenderOptions, 'queries'>,
  delegate = renderLight
): RenderResult & { history: History } {
  const history = createHistory(createMemorySource(route))
  return {
    ...delegate(
      <LocationProvider history={history}>{ui}</LocationProvider>,
      options
    ),
    history,
  }
}
// re-export everything
export * from '@testing-library/react'
// override render method
export { renderLight as render }
export { userEvent }
