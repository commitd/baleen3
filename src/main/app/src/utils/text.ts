/*-
 * #%L
 * Baleen 3
 * %%
 * Copyright (C) 2020 Dstl
 * %%
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * #L%
 */
export const javaClassToJavaName = (javaName: string): string => {
  const split = javaName.split('.')
  const lastValue = split.pop()
  return lastValue === undefined ? '' : lastValue
}

export const javaNameToReadable = (javaName: string): string => {
  const separated = javaName
    .replace(/([A-Z][A-Z])/g, ' $1')
    .replace(/([A-Z][a-z$])/g, ' $1')
    .replace(/[$]/g, '')
  return separated.trim()
}

export const javaToReadable = (input = ''): string =>
  javaNameToReadable(javaClassToJavaName(input))

export const artifactToReadable = (artifact: string): string => {
  const reduced = artifact.replace('annot8-components-', '')
  const split = reduced.split('-')
  return split.map((value) => value[0].toUpperCase() + value.slice(1)).join(' ')
}

export const parseSearch = (search: string): string[] =>
  search
    .split(/[,;]\s*/)
    .map(standardize)
    .filter((term) => term !== '')

export const standardize = (s: string): string =>
  s.trim().toLowerCase().replace(/\s+/g, ' ')
