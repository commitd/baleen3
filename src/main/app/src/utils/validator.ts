/*-
 * #%L
 * Baleen 3
 * %%
 * Copyright (C) 2020 Dstl
 * %%
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * #L%
 */
import Ajv from 'ajv'
import { PipelineDescriptor } from '../types'

const ajv = new Ajv()
const validate = ajv.compile({
  $schema: 'http://json-schema.org/draft-07/schema',
  $id: 'https://github.com/dstl/baleen3/ui/pipeline.json',
  type: 'object',
  title: 'Pipeline',
  description: 'JSON document describing a Baleen pipeline.',
  required: ['sources', 'processors'],
  additionalProperties: false,
  properties: {
    name: {
      type: 'string',
      title: 'Name',
      description: 'The name of the pipeline.',
      default: '',
    },
    description: {
      type: 'string',
      title: 'Description',
      description: 'A description of the pipeline.',
      default: '',
    },
    errorConfiguration: {
      type: 'object',
      title: 'Error Configuration',
      description: 'Error configuration of the pipeline',
      properties: {
        onItemError: {
          enum: ['DISCARD_ITEM', 'REMOVE_PROCESSOR', 'IGNORE'],
          title: 'Item Error',
          description: 'What to do on an Item error',
        },
        onProcessorError: {
          enum: ['DISCARD_ITEM', 'REMOVE_PROCESSOR', 'IGNORE'],
          title: 'Processor Error',
          description: 'What to do on a Processor error',
        },
        onSourceError: {
          enum: ['REMOVE_SOURCE', 'IGNORE'],
          title: 'Source Error',
          description: 'What to do on a Source error',
        },
      },
    },
    sources: {
      type: 'array',
      title: 'Pipeline Sources',
      description: 'The sources for the pipeline.',
      default: [],
      additionalItems: true,
      items: {
        anyOf: [
          {
            type: 'object',
            title: 'Source',
            default: {},
            examples: [{}],
            required: [],
            additionalProperties: true,
            properties: {},
          },
        ],
        $id: '#/properties/sources/items',
      },
    },
    processors: {
      $id: '#/properties/processors',
      type: 'array',
      title: 'Pipeline Processors',
      description: 'The processors that make up the pipeline, in order.',
      default: [],
      additionalItems: true,
      items: {
        anyOf: [
          {
            $id: '#/properties/processors/items/anyOf/0',
            type: 'object',
            title: 'Processor',
            default: {},
            examples: [
              {
                'io.annot8.components.geo.processors.Mgrs': {
                  name: 'Geo',
                  settings: {
                    ignoreDates: true,
                  },
                },
              },
            ],
            required: [],
            additionalProperties: true,
            properties: {},
          },
        ],
        $id: '#/properties/processors/items',
      },
    },
  },
})

export const readUploadedFileAsText = async (
  inputFile: File
): Promise<string | ArrayBuffer | null> => {
  const temporaryFileReader = new FileReader()

  return new Promise((resolve, reject) => {
    temporaryFileReader.onerror = (): void => {
      temporaryFileReader.abort()
      reject(new DOMException('Unable to read file.'))
    }

    temporaryFileReader.onload = (): void => {
      resolve(temporaryFileReader.result)
    }
    temporaryFileReader.readAsText(inputFile)
  })
}

export const validateJson = (data: string): unknown => {
  try {
    // eslint-disable-next-line @typescript-eslint/no-unsafe-return
    return JSON.parse(data)
  } catch (error) {
    throw Error('Can not parse the file.')
  }
}

export const validatePipeline = (
  pipeline: unknown
): Partial<PipelineDescriptor> => {
  const valid = validate(pipeline)
  if (valid !== true) {
    let message = 'JSON not valid:'
    if (!(validate.errors == null)) {
      validate.errors.forEach((error: Ajv.ErrorObject): void => {
        message += error.message
      })
    }
    throw Error(message)
  }
  return pipeline as Partial<PipelineDescriptor>
}

export const validatePipelineFile = async (
  file: File
): Promise<Partial<PipelineDescriptor>> => {
  const fileContent = await readUploadedFileAsText(file)
  if (typeof fileContent === 'string') {
    return validatePipeline(validateJson(fileContent))
  } else {
    throw Error('File incorrect data.')
  }
}
