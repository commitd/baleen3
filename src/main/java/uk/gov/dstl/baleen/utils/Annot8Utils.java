package uk.gov.dstl.baleen.utils;

/*-
 * #%L
 * Baleen 3
 * %%
 * Copyright (C) 2020 Dstl
 * %%
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * 
 *      http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * #L%
 */

import io.annot8.api.pipelines.NoOpOrderer;
import io.annot8.api.pipelines.PipelineOrderer;
import uk.gov.dstl.baleen.exceptions.BadRequestException;
import uk.gov.dstl.baleen.exceptions.ComponentNotFoundException;

public class Annot8Utils {
  public static PipelineOrderer getOrderer(String ordererClass){
    //Special case for NoOpOrderer, as it is a singleton
    if(NoOpOrderer.class.getName().equals(ordererClass))
      return NoOpOrderer.getInstance();

    try {
      Class<? extends PipelineOrderer> clazz = (Class<? extends PipelineOrderer>) Class.forName(ordererClass);
      return clazz.getConstructor().newInstance();
    }catch (ClassNotFoundException e) {
      throw new ComponentNotFoundException(ordererClass);
    }catch (ClassCastException e){
      throw new BadRequestException("Class is not an instance of PipelineOrderer");
    }catch (NoSuchMethodException e){
      throw new BadRequestException("Unable to instantiate Orderer - must have a no argument constructor");
    }catch (Exception e){
      throw new BadRequestException("Unable to instantiate Orderer", e);
    }
  }
}
