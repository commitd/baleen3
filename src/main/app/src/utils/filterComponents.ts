/*-
 * #%L
 * Baleen 3
 * %%
 * Copyright (C) 2020 Dstl
 * %%
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * #L%
 */
import { ComponentInfo, ComponentMap } from '../types'
import { parseSearch, standardize } from './text'

export const filterComponents = (
  map: ComponentMap,
  search: string
): Array<ComponentInfo> => {
  const values = Object.values(map)
  if (search.trim() === '') {
    return values
  }

  const searchTerms = parseSearch(search)

  return values.filter(
    (value) =>
      filterBy('artifact', value, searchTerms) ||
      filterBy('description', value, searchTerms) ||
      filterByName(value, searchTerms) ||
      filterByTags(value, searchTerms)
  )
}

function filterBy(
  key: keyof Omit<ComponentInfo, 'tags'>,
  value: ComponentInfo,
  terms: string[]
): boolean {
  // false positive security/detect-object-injection
  // eslint-disable-next-line
  const property = value[key]
  if (property === undefined) {
    return false
  }
  return terms.some((term) => standardize(property).includes(term))
}

function filterByName(value: ComponentInfo, terms: string[]): boolean {
  const property = standardize(value.name)
  return terms.some((term) => property.includes(term))
}

function filterByTags(value: ComponentInfo, terms: string[]): boolean {
  const tags = value.tags?.map((tag) => tag.toLowerCase())
  return terms.some((term) => tags?.includes(term))
}
