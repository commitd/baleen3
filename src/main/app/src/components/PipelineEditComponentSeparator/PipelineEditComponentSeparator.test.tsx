/*-
 * #%L
 * Baleen 3
 * %%
 * Copyright (C) 2020 Dstl
 * %%
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * #L%
 */
import React from 'react'
import { PipelineEditComponentSeparator } from '.'
import { act, render, renderDark, userEvent } from '../../utils/test'

it('renders without error', () => {
  const { asFragment } = render(
    <PipelineEditComponentSeparator
      type="Processor"
      onInsert={jest.fn()}
      isDragging={false}
    />
  )
  expect(asFragment()).toMatchSnapshot()
})

it('renders dark without error', () => {
  const { asFragment } = renderDark(
    <PipelineEditComponentSeparator
      type="Processor"
      onInsert={jest.fn()}
      isDragging={false}
    />
  )
  expect(asFragment()).toMatchSnapshot()
})

it('Can open dialog and close', async () => {
  const { getByRole, findByText } = renderDark(
    <PipelineEditComponentSeparator
      type="Processor"
      onInsert={jest.fn()}
      isDragging={false}
    />
  )
  act(() => {
    userEvent.click(getByRole('button', { name: /Insert Processors/i }))
  })

  expect(await findByText(/Select Processor/)).not.toBeEmpty()

  act(() => {
    userEvent.click(getByRole('button', { name: /cancel/i }))
  })
})
