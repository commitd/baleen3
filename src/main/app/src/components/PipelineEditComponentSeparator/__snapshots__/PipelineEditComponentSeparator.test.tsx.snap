// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders dark without error 1`] = `
<DocumentFragment>
  <div>
    <button
      aria-label="Insert Processors"
      class="MuiButtonBase-root"
      tabindex="0"
      title="Insert Processors"
      type="button"
    >
      <div
        class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
      >
        <div
          class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root Styled(Styled(MuiBox))-root Styled(Styled(MuiBox))-root"
        />
      </div>
      <div
        class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
        style="opacity: 0; visibility: hidden;"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
      </div>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;

exports[`renders without error 1`] = `
<DocumentFragment>
  <div>
    <button
      aria-label="Insert Processors"
      class="MuiButtonBase-root"
      tabindex="0"
      title="Insert Processors"
      type="button"
    >
      <div
        class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
      >
        <div
          class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root Styled(Styled(MuiBox))-root Styled(Styled(MuiBox))-root"
        />
      </div>
      <div
        class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
        style="opacity: 0; visibility: hidden;"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
      </div>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
